net/respondd-wifi/src/respondd.c

```diff
@@ -0,0 +1,104 @@
+#include <string.h>
+#include <stdio.h>
+#include <json-c/json.h>
+#include <respondd.h>
+
+#include <linux/nl80211.h>
+
+#include "clients.h"
+#include "neighbours.h"
+#include "ifaces.h"
+
+static struct json_object *respondd_provider_statistics(void) {
+	struct json_object *result, *clients;
+	struct iface_list *ifaces;
+	int wifi24 = 0, wifi5 = 0, clients_count;
+
+	result = json_object_new_object();
+	if (!result)
+		return NULL;
+
+	clients = json_object_new_object();
+	if (!clients) {
+		json_object_put(wireless);
+		json_object_put(result);
+		return NULL;
+	}
+
+	ifaces = get_ifaces();
+	while (ifaces != NULL) {
+		if(ifaces->type != NL80211_IFTYPE_AP)
+			goto next_statistics;
+		
+		clients_count = 0;
+		get_client_counts(&clients_count, ifaces->ifx);
+		if (ifaces->frequency < 5000)
+			wifi24 += clients_count;
+		if (ifaces->frequency > 5000)
+			wifi5 += clients_count;
+next_statistics: ;
+		void *freeptr = ifaces;
+		ifaces = ifaces->next;
+		free(freeptr);
```

Comment:
The memory handling in both loops processing the get_ifaces() output is quite ugly. I think a for_each_iface() taking a callback function would be much nicer, as we could avoid building a list of all interfaces in memory.